server/src/test/java/org/elasticsearch/gateway/GatewayServiceTests.java

```diff
@@ -69,4 +97,23 @@ public void testDeprecatedSettings() {
         assertSettingDeprecationsAndWarnings(new Setting<?>[] {GatewayService.RECOVER_AFTER_MASTER_NODES_SETTING });
     }
 
+    public void testRecoverStateUpdateTask() throws Exception {
+        GatewayService service = createService(Settings.builder());
+        ClusterStateUpdateTask clusterStateUpdateTask = service.new RecoverStateUpdateTask();
+        String nodeId = randomAlphaOfLength(10);
+        DiscoveryNode masterNode = DiscoveryNode.createLocal(settings(Version.CURRENT)
+                .put(Node.NODE_MASTER_SETTING.getKey(), true).build(),
+            new TransportAddress(TransportAddress.META_ADDRESS, 9300), nodeId);
+        ClusterState stateWithBlock = ClusterState.builder(ClusterName.DEFAULT)
+            .nodes(DiscoveryNodes.builder().localNodeId(nodeId).masterNodeId(nodeId).add(masterNode).build()).
+                blocks(ClusterBlocks.builder().addGlobalBlock(STATE_NOT_RECOVERED_BLOCK).build()).build();
+
+        ClusterState recoveredState = clusterStateUpdateTask.execute(stateWithBlock);
+        assertNotEquals(recoveredState, stateWithBlock);
+        assertNotEquals(recoveredState.blocks().global(ClusterBlockLevel.METADATA_WRITE), hasItem(STATE_NOT_RECOVERED_BLOCK));
```

Comment:
This assertion is vacuous, I think you mean this:

```suggestion
        assertThat(recoveredState.blocks().global(ClusterBlockLevel.METADATA_WRITE), not(hasItem(STATE_NOT_RECOVERED_BLOCK)));
```